analyzers/src/SonarAnalyzer.CSharp/Metrics/CSharpCognitiveComplexityMetric.cs

```diff
@@ -139,11 +162,10 @@ public override void VisitCatchClause(CatchClauseSyntax node)
             public override void VisitInvocationExpression(InvocationExpressionSyntax node)
             {
                 var identifierNameSyntax = node.Expression as IdentifierNameSyntax;
-                if (State.CurrentMethod != null &&
-                    identifierNameSyntax != null &&
-                    node.HasExactlyNArguments(State.CurrentMethod.ParameterList.Parameters.Count) &&
-                    string.Equals(identifierNameSyntax.Identifier.ValueText,
-                        State.CurrentMethod.Identifier.ValueText, StringComparison.Ordinal))
+                if (State.CurrentMethod != null
+                    && identifierNameSyntax != null
```

Comment:
`&& node.Expression is IdentifierNameSyntax identifierNameSyntax` and we can remove the first method line